blocks/address/src/app.tsx

```diff
@@ -0,0 +1,677 @@
+                  [addressIdKey]: addressId,
+                },
+                linkData: {
+                  leftEntityId: entityId,
+                  rightEntityId: fileEntityId,
+                },
+              },
+            });
+          }
+        }
+      });
+  };
+
+  const updateAddress = async (address?: Address) => {
+    if (readonly || !address) {
+      return;
+    }
+
+    const {
+      addressLocality,
+      addressRegion,
+      postalCode,
+      streetAddress,
+      addressCountry,
+      fullAddress,
+    } = address;
+
+    const addressProperties = {
+      [localityKey]: addressLocality,
+      [regionKey]: addressRegion,
+      [postalCodeKey]: postalCode,
+      [streetKey]: streetAddress,
+      [countryKey]: addressCountry,
+      [fullAddressKey]: fullAddress,
+    };
+
+    const createAddressEntityResponse = await (!addressEntity
+      ? graphService?.createEntity({
+          data: {
+            entityTypeId: addressTypeId,
+            properties: addressProperties,
+          },
+        })
+      : graphService?.updateEntity({
+          data: {
+            entityId: addressEntity.metadata.recordId.entityId,
+            entityTypeId: addressEntity.metadata.entityTypeId,
+            properties: addressProperties,
+          },
+        }));
+
+    const addressEntityId =
+      createAddressEntityResponse?.data?.metadata.recordId.entityId;
+
+    if (addressEntityId) {
+      if (!addressLinkEntity) {
+        await graphService?.createEntity({
+          data: {
+            entityTypeId:
+              "https://alpha.hash.ai/@luisbett/types/entity-type/address-link/v/1",
+            properties: {},
+            linkData: {
+              leftEntityId: entityId,
+              rightEntityId: addressEntityId,
+            },
+          },
+        });
+      }
+    }
+  };
+
+  const resetBlock = async () => {
+    selectAddress();
+    await graphService?.updateEntity({
+      data: {
+        entityId,
+        entityTypeId,
+        properties: {},
+      },
+    });
+
+    // Remove the address link and all image links
+    for (const { linkEntity } of linkedEntities) {
+      if (linkEntity[0]) {
+        await graphService?.deleteEntity({
+          data: {
+            entityId: linkEntity[0].metadata.recordId.entityId,
+          },
+        });
+      }
+    }
+  };
+
+  useEffect(() => {
+    if (selectedAddress) {
+      if (addressId !== selectedAddress.addressId) {
+        updateAddress(selectedAddress);
+        updateBlockAddress(selectedAddress);
+      }
+    } else {
+      resetBlock();
+    }
+  }, [selectedAddress]);
+
+  useEffect(() => {
+    if (mapFile) {
+      uploadMap(mapFile);
+    }
+  }, [mapFile]);
+
+  return (
+    <ThemeProvider theme={theme}>
+      <Box
+        ref={blockRootRef}
+        sx={{ display: "inline-block" }}
+        onMouseEnter={() => setHovered(true)}
+        onMouseLeave={() => setHovered(false)}
+      >
+        <Fade
```

Comment:
This shouldn't appear in `readonly` mode

Reply:
Hid it in readonly mode in 38c4a19ea9d64daea84706f708acccffd3e7ebc9